[Logs UI] Allow dragging of the log minimap to change visible logs

Zacqary opened the pull request:
## Summary

Closes #36119 

![dragmove](https://user-images.githubusercontent.com/1445834/62739718-275f5600-b9fb-11e9-9429-bc5716242c82.gif)

Adds the ability to click and drag on the log minimap on the right side of the logs UI to change the visible window of logs. You can click and drag the time ruler, and click on the minimap itself to jump to a point in time.

Implements visual redesigns from @hbharding: https://www.figma.com/file/Rk8Cgr7VlwiWVMBrJGNGXN8n/Logs-UI?node-id=0%3A1


### Checklist

Use ~~strikethroughs~~ to remove checklist items you don't feel are applicable to this PR.

- [ ] This was checked for cross-browser compatibility, [including a check against IE11](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility)
- ~[ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)~
- ~[ ] [Documentation](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#writing-documentation) was added for features that require explanation or tutorials~
- ~[ ] [Unit or functional tests](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#cross-browser-compatibility) were updated or added to match the most common scenarios~
- ~[ ] This was checked for [keyboard-only and screenreader accessibility](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Accessibility#Accessibility_testing_checklist)~

### For maintainers

- [ ] This was checked for breaking API changes and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)
- [ ] This includes a feature addition or change that requires a release note and was [labeled appropriately](https://github.com/elastic/kibana/blob/master/CONTRIBUTING.md#release-notes-process)

elasticmachine commented:
Pinging @elastic/infra-logs-ui

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/5145/)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/6345/)

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/6407/)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/9237/)

jasonrhodes commented:
@Zacqary what's the status of this, is it still valid for review or does it need more work? Do you know if anyone from our team started looking at it at all?

Zacqary commented:
I think this needs more work before being merged but I'd like someone from the team to play with it and at least comment on whether the UX is going in the right direction. Haven't gotten any feedback on that yet.

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/9850/)

jasonrhodes commented:
So @Zacqary fwiw I really like how this works and this makes the minimap sort of work how my brain expected it to work in the first place. The main thing I notice while playing with it is that it temporarily reverts back to the shape it was in before drag, I think while the new data loads, and then snaps back to where I left it, which sort of breaks the illusion. If there was a way to keep it where you drop it until the new data comes that'd be best, or else some kind of temp loading state. 

Those are my thoughts on general direction. Do you have other concerns about the work here?

Zacqary commented:
@jasonrhodes Thanks for the feedback! I think this works a lot better than it did when I initially opened my PR after the `jumpToTimestamp` accuracy fix. The reason I wanted to shop it around is because getting the minimap to maintain its post-drag position before data load is a non-trivial amount of work, so I just wanted to be sure this was going in the right UX direction before polishing that.

Zacqary commented:
Added some overscan so the user never runs out of TimeRuler. However, because the DensityMap is only rendered with available downloaded data, it's sometimes possible to drag the map beyond the bounds of available dataviz. So I added these stripes for cases like that:

![overscan](https://user-images.githubusercontent.com/1445834/62580570-0ae0e380-b86c-11e9-8b3b-5734f14e73cb.gif)

When you release the mouse button, additional map usually shows up in the striped area after new data is loaded.

@hbharding any thoughts on that visual?

Zacqary commented on x-pack/legacy/plugins/infra/public/components/logging/log_minimap/log_minimap.tsx:
```diff
@@ -152,7 +245,18 @@ const TimeCursor = euiStyled.line`
 `;
 
-const MinimapWrapper = euiStyled.svg`
+const MinimapWrapper = euiStyled.svg<any>`
```
Trying to figure out a better way to get this to pass typechecking without resorting to `<any>`

Zacqary replied:
As it turns out that's just how `<>` works in TypeScript lol

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/11178/)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/11324/)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/11366/)

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/11586/)

snide commented on x-pack/legacy/plugins/infra/public/components/logging/log_minimap/log_minimap.tsx:
```diff
@@ -135,8 +250,9 @@ export class LogMinimap extends React.Component<LogMinimapProps, LogMinimapState
-  fill: ${props => props.theme.eui.euiColorLightestShade};
+const DragTargetArea = euiStyled.rect`
+  fill: transparent;
+  cursor: move;
```
You might want to look into using `grab` (`:hover`) and `grabbing` (`:active`) as your cursor styles for these.

https://www.w3.org/TR/css-ui-3/#valdef-cursor-grab

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/11636/)

Zacqary commented:
@hbharding Do you think the time ruler feels "grabby" enough? Like would the user know to mouse over it and discover the grab cursor? I'm not sure

elasticmachine commented:
## :broken_heart: Build Failed
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/11767/)

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/11780/)

jasonrhodes commented:
I think the diagonal stripes might attract _too much_ attention, maybe just a dark grey background for that area is plenty? I haven't seen it that way so maybe it's confusing, but traffic warning stripes feel a little too "alarming" for the use case, imo...

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/11815/)

elasticmachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pull-request/11985/)
